core/src/starknet_client/client.rs

```diff
@@ -139,7 +139,7 @@ impl StarknetClientConfig {
     ///
     /// Only when this is enabled may the client write transactions to the sequencer
     pub fn account_enabled(&self) -> bool {
-        self.starknet_pkey.is_some() && self.starknet_account_address.is_some()
+        self.starknet_pkeys.is_some() && self.starknet_account_addresses.is_some()
```

Comment:
Related to the above, if we take the account info as `Vec<String>`, we just change this check to `!self.starknet_pkeys.is_empty() && ...`